### PR TITLE
Add sublayers as group to current layer/group

### LINKED ISSUE #48694: "Add to group" option in sub layer window always add at top of layer tree
### What is the bug or the crash?

When option "Add to group" is checked in sub layer window, the group is added at the top of layer tree.
Could it be added inside current selected group or beside current selected layer ?

If the option is unchecked, layers are added inside current selected group or beside current selected layer.

### Steps to reproduce the issue

1. drap and drop a layer with multiple sub layer (like a gpkg with multiple layers)
2. Check "add to group"
3. Add layer
4. See the group added at the top of layer tree

### Versions

3.24.3

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by elpaso**
Fixes #48694

### CODE REVIEW

File: src/app/qgisapp.cpp [hunk 1]
```diff
@@ -5725,7 +5725,38 @@ QList< QgsMapLayer * > QgisApp::addSublayers( const QList<QgsProviderSublayerDet
...
   if ( !groupName.isEmpty() )
   {
-    group = QgsProject::instance()->layerTreeRoot()->insertGroup( 0, groupName );
+    int index { 0 };
+    QgsLayerTreeNode *currentNode { mLayerTreeView->currentNode() };
+    if ( currentNode && currentNode->parent() )
+    {
+      if ( QgsLayerTree::isGroup( currentNode ) )
+      {
+        group = static_cast<QgsLayerTreeGroup *>( currentNode )->insertGroup( 0, groupName );
```
Comment (nyalldawson) [inline src/app/qgisapp.cpp:5734]:
```suggestion
        group = qobject_cast<QgsLayerTreeGroup *>( currentNode )->insertGroup( 0, groupName );
```

File: src/app/qgisapp.cpp [hunk 2]
```diff
@@ -5725,7 +5725,38 @@ QList< QgsMapLayer * > QgisApp::addSublayers( const QList<QgsProviderSublayerDet
...
+      }
+      else if ( QgsLayerTree::isLayer( currentNode ) )
+      {
+        const auto children { currentNode->parent()->children() };
```
Comment (nyalldawson) [inline src/app/qgisapp.cpp:5738]:
```suggestion
        const QList<QgsLayerTreeNode *> currentNodeSiblings { currentNode->parent()->children() };
```

File: src/app/qgisapp.cpp [hunk 3]
```diff
@@ -5725,7 +5725,38 @@ QList< QgsMapLayer * > QgisApp::addSublayers( const QList<QgsProviderSublayerDet
...
+        int nodeIdx { 0 };
+        for ( const auto &child : std::as_const( children ) )
```
Comment (nyalldawson) [inline src/app/qgisapp.cpp:5740]:
```suggestion
        for ( const QgsLayerTreeNode* child : std::as_const( currentNodeSiblings ) )
```

File: src/app/qgisapp.cpp [hunk 4]
```diff
@@ -5725,7 +5725,38 @@ QList< QgsMapLayer * > QgisApp::addSublayers( const QList<QgsProviderSublayerDet
...
+        {
+          nodeIdx++;
+          if ( child == currentNode )
+          {
+            index = nodeIdx;
+            break;
+          }
+        }
```
Comment (nyalldawson) [inline src/app/qgisapp.cpp:5748]: This logic ideally belongs in something like QgsLayerTreeUtils::findNodeIndexInGroup()

**Review by nyalldawson — APPROVED** [review]

**Comment by elpaso** [comment]
unrelated test failure

**Pull request merged**